src/main/java/org/folio/bulkops/processor/marc/MarcFlowCommitProcessor.java

```diff
@@ -0,0 +1,179 @@
+package org.folio.bulkops.processor.marc;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import static org.apache.commons.io.FilenameUtils.getBaseName;
+import static org.folio.bulkops.domain.bean.Instance.INSTANCE_HRID;
+import static org.folio.bulkops.util.Constants.CHANGED_CSV_PATH_TEMPLATE;
+import static org.folio.bulkops.util.Constants.CHANGED_MARC_PATH_TEMPLATE;
+import static org.folio.bulkops.util.Constants.ENRICHED_PREFIX;
+import static org.folio.bulkops.util.Constants.LINE_BREAK;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.opencsv.CSVReaderBuilder;
+import com.opencsv.exceptions.CsvDataTypeMismatchException;
+import com.opencsv.exceptions.CsvRequiredFieldEmptyException;
+import com.opencsv.exceptions.CsvValidationException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.collections4.CollectionUtils;
+import org.folio.bulkops.client.RemoteFileSystemClient;
+import org.folio.bulkops.domain.bean.BulkOperationsEntity;
+import org.folio.bulkops.domain.bean.ExtendedInstance;
+import org.folio.bulkops.domain.bean.Instance;
+import org.folio.bulkops.domain.converter.BulkOperationsEntityCsvWriter;
+import org.folio.bulkops.domain.entity.BulkOperation;
+import org.folio.bulkops.exception.ConverterException;
+import org.folio.bulkops.util.CSVHelper;
+import org.folio.bulkops.util.UnifiedTableHeaderBuilder;
+import org.marc4j.MarcStreamReader;
+import org.marc4j.MarcStreamWriter;
+import org.springframework.stereotype.Service;
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.SequenceInputStream;
+import java.io.StringWriter;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class MarcFlowCommitProcessor {
```

Comment:
This should implements interface. New or existing interface should be implemented/reused.